### ISSUE TITLE
(aws-mediapackage): attrHlsIngest property missing on CfnChannel

### ISSUE THREAD

**Issue by jkueh** [aws/aws-cdk#12462]
Hello!

Apologies if this should be under another issue type.

The `attrHlsIngest` property appears to be missing on `CfnChannel`.

Looking at some other L1 constructs, I can see that they have their properties that would typically be reachable via `Fn::GetAtt` in vanilla Cloudformation, e.g.:

```json
"Fn::GetAtt" : [ "myMediaPackageChannel" , "HlsIngest" ]
```

For example an S3 Bucket (`CfnBucket`) has:

* `attrArn`
* `attrDomainName`, `attrDualStackDomainName`, `attrRegionalDomainName`
* `attrWebsiteUrl`

(from https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-s3.CfnBucket.html#properties)

Comparing this to the MediaPackage `CfnChannel` class, however:
* `attrArn`

(from https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-mediapackage.CfnChannel.html#properties)

Going through the type definition for CfnChannel also shows that while there isn't a property for the HLS ingest attributes, there does appear to be an `HlsIngestProperty` interface, which suggests I may be doing this wrong.

### Reproduction Steps

1. Install aws-cdk
2. Create MediaPackage channel, e.g.:
```typescript
import * as mediapackage from '@aws-cdk/aws-mediapackage';

const channel = new mediapackage.CfnChannel(this, "test-channel", {
  id: "test-channel",
});
```
3. Attempt to access properties for the HLS ingest endpoints

### What did you expect to happen?

My IDE to offer `attrHlsIngest` and all its corresponding data as a suggested completion, and for this property to be available.

### What actually happened?

I couldn't work out a way to access these properties from within the CDK.

I accept that I might be doing this wrong!

### Environment

  - **CDK CLI Version  :** 1.83.0
  - **Framework Version:**
  - **Node.js Version:** 14.15.4
  - **OS               :** Windows
  - **Language (Version):** all

### Other

--- 

This is :bug: Bug Report

**Comment by skinny85** [comment]
Hey @jkueh ,

you're right that `CfnChannel` in the CDK is missing the `HlsIngest` attribute. It's actually not a coincidence, but a conscious decision on our part - we removed it deliberately [here](https://github.com/aws/aws-cdk/blob/422dc8e9d50105af4e710d409a4f301079d43f3f/packages/%40aws-cdk/cfnspec/spec-source/680_MediaPackage_Channel_patch.json#L1-L15).

The reason we did that is because `HlsIngest` returns a complex type, and there is no way currently in CloudFormation to refer to a property of something returned from `Fn::GetAtt` that is  a complex type. So, we don't see value in these attributes, and our generation of CloudFormation resources doesn't handle them.

If you still want to have the expression `{ 'Fn::GetAtt': ['test-channel', 'HlsIngest'] }`, you can always create it directly yourself:

```ts
import * as mediapackage from '@aws-cdk/aws-mediapackage';

        const channel = new mediapackage.CfnChannel(this, "test-channel", {
            id: "test-channel",
        });
        new CfnOutput(this, 'Output', {
            value: channel.getAtt('HlsIngest').toString(),
        });
```

(Your example code was for `aws-medialive`, not `aws-mediapackage` - I assume that's just a typo)

Can I ask - what do you need the attribute for?

Thanks,
Adam

**Comment by jkueh** [comment]
Hi @skinny85!

That makes sense. And yes, `medialive` was on the brain, so I've fixed that typo - Thanks :)
I'm attempting to tie MediaLive to MediaPackage for delivery via Cloudfront.

I was poking around the HlsIngest payload to try and get more information out of the MediaPackage channel.
If memory serves me right, I think I was attempting to connect MediaLive to MediaPackage via HLS as I was having some issues using the CDK's L1 constructs (which have since been resolved)

As you mentioned, it's a complex type so it won't let me use the `HlsIngest` directly, however if I try to access a given attribute:

```typescript
new cdk.CfnOutput(this, 'hlsingest', {
  value: mPackageChannel.getAtt('HlsIngest.ingestEndpoints[0].Url').toString(),
})
```

This doesn't seem to make Cloudformation too happy either - I've tried capitalising the `i` in `ingestEnpoints`, and index 1 instead of index 0.

Either way, at this point I have a working MediaLive -> MediaPackage implementation (or so the console tells me), so I'm happy to close this issue :)

Thanks!
--Jordan

**Comment by skinny85** [comment]
Like I said - there's no way currently in CloudFormation to take out the property `IngestEndpoints` from the value returned from `{ "Fn::GetAtt": ["myMediaPackageChannel" , "HlsIngest" ] }`, so it's not something we can offer in the CDK.

Glad you got yourself unblocked, let us know if there's anything else we can help you with on this issue!
